### PR TITLE
GC+/additional first slice tests

### LINKED ISSUE #4788: GC+: Add additional checks in integration tests
Add a check that first slice is select correctly in the following scenarios:
1. New structured data - first slice (according to order)
2. Only old structured data - empty string
3. With old staged objects format with keys that are smaller than slice keys - first valid slice (according to order)
4. Only staged objects with no new structured data - empty string

### LINKED ISSUE #4869: GC+: Add additional first slice checks

### PR DESCRIPTION
**Pull request by eden-ohana**
closes #4869

### CODE REVIEW

File: clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala [hunk 1]
```diff
@@ -148,25 +148,63 @@ class UncommittedGarbageCollectorSpec
           UncommittedGarbageCollector.getFirstSlice(dataDF, repo) should be("")
         })
       }
+
+      it("should return empty first slice with only staged objects") {
+        withSparkSession(_ => {
+          val dataDir = new File(dir.toFile, "data")
+          dataDir.mkdir()
+          val legacySlice = repo + "_legacy_physical:address_path"
```
Comment (itaiad200) [inline clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala:156]: It's not a slice, is it?

**Review by itaiad200 — APPROVED** [review]
Looks like a great addition to the coverage, I'm not sure it covers all 4 bullets in the original issue. Approving, I think @N-o-Z should look at this too.

File: clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala [hunk 2]
```diff
@@ -148,25 +148,63 @@ class UncommittedGarbageCollectorSpec
...
           new File(slice, filename).createNewFile()
           slice = new File(dataDir, regularSlice)
           slice.mkdir()
           new File(slice, filename).createNewFile()
+          slice = new File(dataDir, regularSlice2)
+          slice.mkdir()
+          new File(slice, filename).createNewFile()
 
-          val dataDF = UncommittedGarbageCollector.listObjects(dir.toString,
+          var dataDF = UncommittedGarbageCollector.listObjects(dir.toString,
```
Comment (N-o-Z) [inline clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala:202]: I think you should revert this change - dataDF should be a val and lines 206-7 can be written together as a single statement

File: clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala (hunk 1 above)
Comment (N-o-Z) [inline clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala:156]: suggest to change the name - legacyPath or something similar

**Review by N-o-Z — CHANGES_REQUESTED** [review]
@eden-ohana  Thanks for this addition to the unit tests.
Please note that #4788 talks about adding these checks to the **integration** tests - this is still missing.
I suggest creating a new issue for this PR and leaving #4788 open

File: clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala (hunk 1 above)
Comment (eden-ohana) [reply clients/spark/core/src/test/scala/io/treeverse/gc/UncommittedGarbageCollectorSpec.scala]: done

**Review by N-o-Z — APPROVED** [review]
